File: cpp/869.reordered-power-of-2.cpp
class Solution {
public:
    bool isPow(string &st){
	    if(st[0]=='0')return false;
	    int num = stoi(st);
	    return __builtin_popcount(num) == 1;
    }
    bool reorderedPowerOf2(int n) {
         //61 --> 111101
	 // 10000
	 string num = to_string(n);
	 sort(num.begin(), num.end());
	 do{
	      if(isPow(num))return true;
	 }while(next_permutation(num.begin(), num.end()));
	 return false;
    }
};

File: python/869.reordered-power-of-2.py
class Solution:
    def count(self, n: int) -> list:
        freq = [0]*10
        while n > 0:
            freq[n%10]+=1
            n//=10
        return freq

    def reorderedPowerOf2(self, n: int) -> bool:
        ref = self.count(n)
        for i in range(31):
            if ref == self.count(1<<i):
                return True
        return False

